### DISCUSSION TITLE
light and dark mode through toggling of tailwind configuration

### DISCUSSION THREAD

**Discussion by michaelbe812** [tailwindlabs/tailwindcss#14276] [Help]
Hi all,

I'm setting up some design system using Tailwind CSS. My system should support light and dark mode.

I know from the docs that there are variants for it, given the example:
```
<div class="bg-white dark:bg-slate-800 rounded-lg px-6 py-8 ring-1 ring-slate-900/5 shadow-xl">
  <div>
    <span class="inline-flex items-center justify-center p-2 bg-indigo-500 rounded-md shadow-lg">
      <svg class="h-6 w-6 text-white" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><!-- ... --></svg>
    </span>
  </div>
  <h3 class="text-slate-900 dark:text-white mt-5 text-base font-medium tracking-tight">Writes Upside-Down</h3>
  <p class="text-slate-500 dark:text-slate-400 mt-2 text-sm">
    The Zero Gravity Pen can be used to write in any orientation, including upside-down. It even works in outer space.
  </p>
</div>
```

However this is not the approach I want to do. With plain css I would typically define something like this:

```
:root{
 --color-primary: #000000
 // some more
}

.dark-theme: {
   --color-primary: #ffffff
}

```
and then conditionally apply the `dark-theme` class. 

What I want to achieve is that I e.g. define my `color-primary` in the tailwind config for the default theme and a value for the dark-theme as well. Based on a trigger it should then be able to switch the values accordingly.

I hope it is clear what I want to achieve

**Comment by wongjn** [comment]
See the documentation about defining colors using CSS variables: https://tailwindcss.com/docs/customizing-colors#using-css-variables

**Reply by michaelbe812** [reply]
Thanks for the suggestion. This is not exactly what I want to achieve. I do not want to define any css variables. 

I only want extend my tailwind config and having for each theme a separate config.

E.g. for a light-theme:

```
const colors = require('tailwindcss/colors')
module.exports = {
  content: ['./src/**/*.{html,js}'],
  theme: {
    colors: {
      'primary': colors.indigo["900"]
    },
  },
}

```
and for a dark-theme:


```
const colors = require('tailwindcss/colors')
module.exports = {
  content: ['./src/**/*.{html,js}'],
  theme: {
    colors: {
      'primary': colors.whatever
    },
  },
}

```

So I need the ability to switch config-files.

**Reply by wongjn** [reply]
You could consider using [presets](https://tailwindcss.com/docs/presets).

**Reply by michaelbe812** [reply]
Yes this is something i can try out or with a custom build step. 

However I had the hope that I could switch configs on the fly during runtime somehow 🤔

**Reply by wongjn** [reply]
Tailwind runs at build time. It does not run at run time so switching configurations then would have no effect, as the CSS has already been built by then.
